SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -834,94 +442,32 @@ void SPATIALGDKSERVICES_API FLocalDeploymentManager::TakeSnapshot(UWorld* World,
-			// the wrong info! So let's wait a bit..
-			FTimerHandle TimerHandle;
-			World->GetTimerManager().SetTimer(
-				TimerHandle,
-				[bUseStandardRuntime, OnSnapshotTaken]() {
-					bool bSuccess = false;
-
-					FString NewestSnapshotFilePath;
-					FString SnapshotSearchDirectory;
-
-					// Standard Runtime places the snapshots in SpatialOS logs, while Compatibility Mode Runtime
-					// places them in AppData.
-					if (bUseStandardRuntime)
-					{
-						FSpatialGDKServicesModule& GDKServices =
-							FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
-						FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
-
-						FString SpatialPath = SpatialGDKServicesConstants::SpatialOSDirectory;
-						SnapshotSearchDirectory =
-							SpatialPath + TEXT("logs/localdeployment/") + LocalDeploymentManager->GetLocalRunningDeploymentID();
-
-						IFileManager& FileManager = FFileManagerGeneric::Get();
-
-						// For Standard you need to check which is the latest .snapshot file that was added
-						// inside that running deployment folder.
-						TArray<FString> SnapshotFiles;
-
-						FileManager.FindFiles(SnapshotFiles, *SnapshotSearchDirectory, TEXT("snapshot"));
-
-						FDateTime NewestSnapshotTimestamp = FDateTime::MinValue();
-
-						for (const FString& SnapshotFile : SnapshotFiles)
-						{
-							FString SnapshotFilePath = SnapshotSearchDirectory + TEXT("/") + SnapshotFile;
-							FDateTime SnapshotFileTimestamp = FileManager.GetTimeStamp(*SnapshotFilePath);
-							if (SnapshotFileTimestamp > NewestSnapshotTimestamp)
-							{
-								NewestSnapshotTimestamp = SnapshotFileTimestamp;
-								NewestSnapshotFilePath = SnapshotFilePath;
-							}
-						}
-					}
-					else
-					{
-						// Compatibility Mode will have the latest created snapshot name in 'latest' file.
-						FString AppDataLocalPath = FPlatformMisc::GetEnvironmentVariable(TEXT("LOCALAPPDATA"));
-						SnapshotSearchDirectory = FString::Printf(TEXT("%s/.improbable/local_snapshots"), *AppDataLocalPath);
-						FString LatestSnapshotInfoPath = FString::Printf(TEXT("%s/latest"), *SnapshotSearchDirectory);
-						FString LatestSnapshot;
-						if (FPaths::FileExists(LatestSnapshotInfoPath)
-							&& FFileHelper::LoadFileToString(LatestSnapshot, *LatestSnapshotInfoPath))
-						{
-							NewestSnapshotFilePath =
-								FString::Printf(TEXT("%s/.improbable/local_snapshots/%s"), *AppDataLocalPath, *LatestSnapshot);
-						}
-					}
-
-					bSuccess = !NewestSnapshotFilePath.IsEmpty();
-
-					if (!bSuccess)
-					{
-						UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed find snapshot file in '%s'"), *SnapshotSearchDirectory);
-					}
-
-					if (OnSnapshotTaken != nullptr)
-					{
-						OnSnapshotTaken(bSuccess, NewestSnapshotFilePath);
-					}
-				},
-				0.5f /* InRate */, false /* InbLoop */);
+			bool bSuccess = false;
```

Comment:
Do we need this variable?

Reply:
I don't think we need this. seems to be used one further down. Might as well just inline it

Reply:
Good catch, refactored.